staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -354,10 +378,39 @@ func (cfgCtlr *configController) digestConfigObjects(newPLs []*flowcontrol.Prior
 			errs = append(errs, errors.Wrap(err, fmt.Sprintf("failed to set a status.condition for FlowSchema %s", fsu.flowSchema.Name)))
 		}
 	}
+	cfgCtlr.addUpdateResult(currResult)
+
 	if len(errs) == 0 {
-		return nil
+		return suggestedDelay, nil
+	}
+	return suggestedDelay, apierrors.NewAggregate(errs)
+}
+
+// shouldDelayUpdate checks to see if a flowschema has been updated too often and returns true if a delay is needed.
+func (cfgCtlr *configController) shouldDelayUpdate(flowSchemaName string) bool {
+	numUpdatesInPastMinute := 0
+	oneMinuteAgo := time.Now().Add(-1 * time.Minute)
+	for _, update := range cfgCtlr.tenMostRecentUpdates {
+		if update.timeUpdated.After(oneMinuteAgo) {
+			continue
+		}
+		if update.updatedItems.Has(flowSchemaName) {
+			numUpdatesInPastMinute++
+		}
+	}
+	if numUpdatesInPastMinute > 5 {
+		return true
+	}
+	return false
+}
+
+// addUpdateResult adds the result and keeps the only the most recent 10. It isn't a ring buffer because I'm lazy and
+// this is small and rate limited
+func (cfgCtlr *configController) addUpdateResult(result updateResult) {
```

Comment:
I think it is true in the current code that there are not concurrent accesses to `cfgCtlr.tenMostRecentUpdates`, but that relies on cross-func analysis.  I would feel more comfortable with comments about this on each relevant func, or using mutexes anyway.